runtime/Go/antlr/atn_config_set.go

```diff
@@ -6,12 +6,13 @@ package antlr
 
 import "fmt"
 
+// ATNConfigSet extends ATNConfig in the form of an unordered set.
 type ATNConfigSet interface {
 	hash() int
-	Add(ATNConfig, *DoubleDict) bool
+	Add(ATNConfig, *doubleDict) bool
```

Comment:
It's unusual to expose unexported names in exported declarations. If Add itself can't be made entirely unexported, the typical design is to leave DoubleDict exported, since the user is responsible for constructing one. I'm not familiar with this code anymore, so perhaps the user has a way to get a doubleDict from somewhere, and knows to plug it in here?